x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/utils/indexing_service.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { http as httpService } from './http_service';
+import { getSavedObjectsClient } from '../../../../kibana_services';
+
+export const getExistingIndexNames = async () => {
+  const indexes = await httpService({
+    url: `/api/index_management/indices`,
+    method: 'GET',
+  });
+  return indexes ? indexes.map(({ name }) => name) : [];
+};
+
+export const createNewIndexAndPattern = async (indexName) => {
+  return await httpService({
+    url: `/api/maps/docSource`,
+    method: 'POST',
+    data: {
+      index: indexName,
+      // Initially set to static mappings
+      mappings: {
+        properties: {
+          coordinates: {
+            type: 'geo_shape',
+          },
+        },
+      },
+    },
+  });
+};
+
+export const addFeatureToIndex = async (indexName, geometry) => {
+  return await httpService({
+    url: `/api/maps/feature`,
+    method: 'POST',
+    data: {
+      index: indexName,
+      data: {
+        coordinates: geometry,
+      },
+    },
+  });
+};
+
+export const getExistingIndexPatternNames = async () => {
+  const indexPatterns = await getSavedObjectsClient()
+    .find({
+      type: 'index-pattern',
+      fields: ['id', 'title', 'type', 'fields'],
+      perPage: 10000,
+    })
+    .then(({ savedObjects }) => savedObjects.map((savedObject) => savedObject.get('title')));
+  return indexPatterns ? indexPatterns.map(({ name }) => name) : [];
+};
+
+export function checkIndexPatternValid(name) {
```

Comment:
Agreed that there's plenty of overlap with File Upload here. We do need to think more about what's worth exposing from File Upload in the long run, but I didn't tackle that in this PR since we're still behind an experimental flag and mostly just leveraging small pieces and utility functions.